### DISCUSSION TITLE
I want a simple API to stream the data for a file download in JavaScript

### DISCUSSION THREAD

**Discussion by aarongustafson** [WebWeWant/webwewant.fyi#158] [Wants]
It&#39;s currently possible to use a Blob URL to initiate the download of data held in memory. This is a useful feature to allow the to user to download data that is generated or processed by JavaScript. However, this only works for smaller amounts of data that can fit in RAM. When retrieving larger amounts of data (e.g., via a WebRTC data channel, or over a WebSocket), there&#39;s no way to allow the user to download the data without first buffering it all and creating a Blob.

I&#39;d like a simple API that allows me to initiate a download with a suggested filename and expected size, triggering a download through the browser&#39;s standard download UI, and allowing me to feed in data as it becomes available. The simplest solution might be to support calling `createObjectURL` with a `ReadableStream`, though it doesn&#39;t look like that would provide a way to specify the expected total size.

<hr>

https://webwewant.fyi/wants/20/

**Comment by guest271314** [comment]
See https://github.com/jimmywarting/StreamSaver.js/issues/13#issuecomment-724419483

**Comment by guest271314** [comment]
Another option is to utilize `QuicTransport`.

**Comment by guest271314** [comment]
Have you solved this?

See https://wicg.github.io/file-system-access/#api-filesystemwritablefilestream
